src/main/kotlin/no/nav/familie/ef/sak/selvstendig/NæringsinntektKontrollService.kt

```diff
@@ -35,67 +37,153 @@ class NæringsinntektKontrollService(
-                secureLogger.info("Kontrollerer person med ident: $personIdent")
+    fun kontrollerInntektForSelvstendigNæringsdrivende(): List<UUID> {
+        if (LeaderClient.isLeader() == false) return emptyList()
 
-                val fagsakOvergangsstønad = fagsakService.finnFagsaker(setOf(personIdent)).firstOrNull { it.stønadstype == StønadType.OVERGANGSSTØNAD } ?: throw RuntimeException("Fant ikke fagsak for overgangsstønad for person: $personIdent")
-                val behandlingId = behandlingService.finnSisteIverksatteBehandling(fagsakOvergangsstønad.id)?.id ?: throw RuntimeException("Fant ingen gjeldende behandling for fagsakId: ${fagsakOvergangsstønad.id}")
-                val tilkjentYtelse = tilkjentYtelseService.hentForBehandling(behandlingId)
+        val fagsakIds = mutableListOf<UUID>()
+        val oppgaver = hentOppgaverForSelvstendigeTilInntektskontroll()
 
-                val antallMåneder = antallMånederMedVedtakForFjoråret(tilkjentYtelse)
-                secureLogger.info("$antallMåneder måneder med vedtak for fagsakId: ${fagsakOvergangsstønad.id} eksternFagsakId: ${fagsakOvergangsstønad.eksternId}")
-                val næringsinntekt = hentFjoråretsNæringsinntekt(fagsakOvergangsstønad.fagsakPersonId)
-
-                if (antallMåneder > 3 && næringsinntekt > INNTEKTSGRENSE_FOR_KONTROLL_AV_AKTIVITET) {
-                    val fjoråretsPersoninntekt = inntektService.hentÅrsinntekt(personIdent, årstallIFjor)
-                    secureLogger.info("Forrige års inntekt for person uten ytelse fra offentlig: $fjoråretsPersoninntekt")
-                    if (fjoråretsPersoninntekt == 0) {
-                        val forventetInntekt = forventetInntektSnitt(tilkjentYtelse)
-                        // secureLogger.info("Beregnet inntekt i snitt for år $årstallIFjor og behandlingId ${behandling.id} er: $forventetInntekt")
-                        if (næringsinntekt > (forventetInntekt * 1.1)) {
-                            secureLogger.info("Har 10% høyere næringsinntekt for person: $personIdent (Næringsinntekt: $næringsinntekt - ForventetInntekt: $forventetInntekt)")
-                            fagsakIds.add(fagsakOvergangsstønad.id)
-                        }
+        val næringsinntektDataForBeregningList = hentNæringsinntektDataForBeregning(oppgaver)
+
+        næringsinntektDataForBeregningList.forEach {
+            if (skalKontrolleres(it.tilkjentYtelse, it.fagsak)) {
+                if (oppfyllerAktivitetsplikt(it.fjoråretsPersonInntekt, it.fjoråretsNæringsinntekt, it.fagsak)) {
+                    if (it.fjoråretsNæringsinntekt >= (it.forventetInntektIFjor * 1.1)) {
+                        secureLogger.info("Har 10% høyere næringsinntekt for person: ${it.personIdent} (Næringsinntekt: ${it.forventetInntektIFjor} - ForventetInntekt: ${it.forventetInntektIFjor})")
+                        fagsakIds.add(it.fagsak.id)
+                        giVarselOmNyVurderingAvInntekt(it.behandlingId, it.personIdent)
```

Comment:
Dette må være et varselbrev

Reply:
Hvis én ting inni her krasjer nå, så blir det vel litt pussig state på alt og man må rekjøre hele greia for hver eneste person.

Er det en idé å lage en task per person her? NyVurdering-task, Beskjed-om-kontrollert-inntekt-task, be-om-regnskap-task? Da vil vi isåfall slippe å måtte rekjøre samme person mange ganger fordi en ting feiler for en annen?

Reply:
Alternativt; alt dette er én task og input til tasken er en oppgave. Da får vi én task per oppgave, som kanskje er like greit..?